w1/models/cat_model.go

```diff
@@ -121,3 +133,24 @@ func EditCat(in EditCatIn, userId string) (Cat, error) {
 
 	return cat, nil
 }
+
+func CatGetById(id string, db *sqlx.DB) (Cat, error) {
+	if db == nil {
+		db = internal.GetDB()
+	}
+
+	cat := Cat{}
+	query := `
+		SELECT 
+			id, name, sex, owner_id, has_matched
+		FROM 
+			cats WHERE id = $1 AND deleted_at IS NULL
+	`
+	err := db.QueryRow(query, id).Scan(&cat.ID, &cat.Name, &cat.Sex, &cat.OwnerID, &cat.HasMatched)
+
+	if errors.Is(err, sql.ErrNoRows) {
+		return Cat{}, CatError{Message: ErrCatNotFound.Error(), Code: http.StatusNotFound}
+	}
+
+	return cat, err
+}
```

Comment:
Since this function just used by match model, just make a single responsibility function in match model to verify:

1. The cat is owner by issuer or receiver user (it's can be done in SQL query).

Reply:
If we verify the cat is owned by either issuer or receiver by sql query, we can't distinguish between whether the cat is **not found** or **not owned** that's why it needed to be done in app side instead.

how about that?

Reply:
Yap, it's a little bit challenging and may make the code hard to read.

1. Query cat by id and owner id
```
id IN (issuerCatId, receiverCatId) AND owner_id (issuerId, receiverUserId)
```
2. If the returned rows < 2, then 404 error

3. In this stage we have only 2 cats, use those for other validation.